modules/edu.gemini.giapi.client/src/main/scala/edu/gemini/giapi/client/package.scala

```diff
@@ -7,31 +7,68 @@ import cats.Applicative
 import cats.effect._
 import cats.implicits._
 import edu.gemini.aspen.giapi.commands.Command
+import edu.gemini.aspen.giapi.status.{StatusHandler, StatusItem}
+import edu.gemini.aspen.giapi.statusservice.{StatusHandlerAggregate, StatusService}
 import edu.gemini.aspen.giapi.util.jms.status.StatusGetter
 import edu.gemini.aspen.gmp.commands.jms.client.CommandSenderClient
 import edu.gemini.giapi.client.commands.CommandResult
 import edu.gemini.jms.activemq.provider.ActiveMQJmsProvider
-import fs2.async
+import fs2.{Stream, async}
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.Duration
 
 package object client {
 
   /**
     * Allowed types according to GIAPI
     */
-  implicit val strItemGetter: ItemGetter[String]    = new ItemGetter[String] {}
-  implicit val doubleItemGetter: ItemGetter[Double] = new ItemGetter[Double] {}
-  implicit val intItemGetter: ItemGetter[Int]       = new ItemGetter[Int]    {}
-  implicit val floatItemGetter: ItemGetter[Float]   = new ItemGetter[Float]  {}
+  implicit val strItemGetter: ItemGetter[String] = new ItemGetter[String] {
+    def value(p: Any): Option[String] = p match {
+      case x: String => Some(x)
+      case _         => None
+    }
+  }
+
+  implicit val doubleItemGetter: ItemGetter[Double] = new ItemGetter[Double] {
+    def value(p: Any): Option[Double] = p match {
+      case x: Double => Some(x)
+      case _         => None
+    }
+  }
+
+  implicit val intItemGetter: ItemGetter[Int] = new ItemGetter[Int] {
+    def value(p: Any): Option[Int] = p match {
+      case x: Int => Some(x)
+      case _      => None
+    }
+  }
+
+  implicit val floatItemGetter: ItemGetter[Float] = new ItemGetter[Float] {
+    def value(p: Any): Option[Float] = p match {
+      case x: Float => Some(x)
+      case _        => None
+    }
+  }
```

Comment:
All this similar looking implementations of `value` could be factored out into a class.

Reply:
I think the matching of `case x: A` wouldn't work. You need tho concrete class there, otherwise I could have done this directly on the method `stream`

Reply:
You can implement this generically but it requires some trickery. See below.